src/app/modules/shared_models/section_model.nim

```diff
@@ -292,6 +290,7 @@ QtObject:
       ModelRole.CanJoin.int,
       ModelRole.Joined.int,
       ModelRole.Spectated.int,
+      ModelRole.Access.int,
```

Comment:
Did we have a way to just say "all ModelRoles"? That would be way better than adding each one manually. It can cause issues like this.

Reply:
I am not aware of any.

Reply:
Should be possible, just by passing an empty array, eg `@[]`; this is further passed on to this nimqml code:
```nim
proc dataChanged*(self: QAbstractItemModel,
                 topLeft: QModelIndex,
                 bottomRight: QModelIndex,
                 roles: openArray[int]) =
  ## Notify the view that the model data changed
  debugMsg("QAbstractItemModel", "dataChanged")
  var copy: seq[cint]
  for i in roles:
    copy.add(i.cint)
  dos_qabstractitemmodel_dataChanged(self.vptr.DosQAbstractItemModel, topLeft.vptr,
                                     bottomRight.vptr, copy[0].addr, copy.len.cint)
```

which then ends up calling the C++ method:
```cpp
void dos_qabstractitemmodel_dataChanged(::DosQAbstractItemModel *vptr,
                                        const ::DosQModelIndex *topLeftIndex,
                                        const ::DosQModelIndex *bottomRightIndex,
                                        int *rolesArrayPtr,
                                        int rolesArrayLength)
{
    auto object = static_cast<QObject *>(vptr);
    auto model = dynamic_cast<DOS::DosIQAbstractItemModelImpl *>(object);
    auto topLeft = static_cast<const QModelIndex *>(topLeftIndex);
    auto bottomRight = static_cast<const QModelIndex *>(bottomRightIndex);
    auto roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
    model->publicDataChanged(*topLeft, *bottomRight, roles);
}
```
where the Qt method signature is:
```cpp
void dataChanged(const QModelIndex &topLeft, const QModelIndex &bottomRight, const QVector<int> &roles = QVector<int>());
```

TLDR, passing an empty `roles` vector means _all_ roles

Reply:
Yeah looks like it, should be easy to fix

Reply:
@jrainville can you quickly try this patch? :)

```diff
--- vendor/DOtherSide/lib/src/DOtherSide.cpp
+++ vendor/DOtherSide/lib/src/DOtherSide.cpp
@@ -1301,7 +1301,10 @@ void dos_qabstractitemmodel_dataChanged(::DosQAbstractItemModel *vptr,
     auto model = dynamic_cast<DOS::DosIQAbstractItemModelImpl *>(object);
     auto topLeft = static_cast<const QModelIndex *>(topLeftIndex);
     auto bottomRight = static_cast<const QModelIndex *>(bottomRightIndex);
-    auto roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
+    QVector<int> roles;
+    if (rolesArrayPtr && rolesArrayLength > 0)
+        roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
+
     model->publicDataChanged(*topLeft, *bottomRight, roles);
 }
```

Reply:
Sorry, it doesn't seem to work either. I'm not sure what's the issue

Reply:
OK, can you please create an issue to track and assign to me? I have some local code to fix but let's deal with it separately

Reply:
Done here: https://github.com/status-im/status-desktop/issues/11830

Reply:
Thanks :)